Codes/Clean Data/getMessageType.R

```diff
@@ -0,0 +1,62 @@
+function(condition,
+         direction,
+         sz,
+         dvol,
+         string1,
+         string2) {
+  
+  
+  
+  #In Terms of Numbers of Messages
+  totalNum <- sum(condition)
+  buyNum <- sum(condition & direction == 1)
+  sellNum <- sum(condition & direction == -1)
+  if (buyNum + sellNum != totalNum) {
+    stop("Buy and Sell numbers don't add up")
+  }
+  
+  #In Terms of Share Volumes of Messages
+  totalSVol <- sum(condition * sz)
+  buySVol <- sum((condition & direction == 1) * sz)
+  sellSVol <- sum((condition & direction == -1) * sz)
+  if (buySVol + sellSVol != totalSVol) {
+    stop("Buy and Sell share volumes don't add up")

```

Comment:
Indeed, you are checking whether directions are all in Set (1, -1). Am I right?

Reply:
Yes, this is just a "sanity check" to make sure the buy + sell = total volume.